### PR TITLE
irmin-pack: only remember commits in `newies'

### PR DESCRIPTION
**Pull request by samoht**
Experimenting with another way to reduce the number of `newies`: only track the new commits and let `Repo.iter` complete the missing ones :-)

But this doesn't pass the tests and I am not sure why..

### CODE REVIEW

**Comment by icristescu** [comment]
It can be that only blobs were added during freeze, in which case, if we don't add them to the newies, they are lost. The issue is that objects are not added "by bulk" for each commit but instead they are added progressively.

**Comment by samoht** [comment]
> It can be that only blobs were added during freeze, in which case, if we don't add them to the newies, they are lost. The issue is that objects are not added "by bulk" for each commit but instead they are added progressively.

So I've modified the lock to make sure freeze are finished only when a "bulk" of writes, corresponding to a high-level commit, is complete. That now seem to pass the tests.

**Comment by samoht** [comment]
Rebased.

File: src/irmin-pack/irmin_pack_layers.ml [hunk 1]
```diff
@@ -775,23 +744,17 @@ struct
...
       let rec copy_newies_to_next_upper t former_offset =
         let newies_limit = blocking_copy_size t.X.Repo.config |> Int64.of_int in
         let offset = X.Repo.offset t in
         if offset -- former_offset >= newies_limit then
-          copy_newies_aux ~with_lock:false t >>= fun () ->
+          copy_newies t >>= fun () ->
           (copy_newies_to_next_upper t offset [@tail])
         else Lwt.return_unit
-
-      (** If there are only a few newies left (less than newies_limit bytes
```
Comment (icristescu) [inline src/irmin-pack/irmin_pack_layers.ml:792]: maybe worthwhile to keep this comment somewhere

File: src/irmin-pack/irmin_pack_layers.ml [hunk 2]
```diff
@@ -726,42 +728,9 @@ struct
           nodes (or contents) not yet attached to a commit (or a node resp.), so
           we have to iter over the graph of commits, iter over the graph of
           nodes, and lastly iter over contents. *)
```
Comment (icristescu) [inline src/irmin-pack/irmin_pack_layers.ml:730]: can you rewrite the comments above to reflect this changes? we no longer need to iter over nodes and contents separately, as it is suggested.

**Review by icristescu — APPROVED** [review]
Thanks, this is really nice!

**Comment by samoht** [comment]
@icristescu I've modified the comments, let me know what you think.

**Comment by icristescu** [comment]
> @icristescu I've modified the comments, let me know what you think.

looks good, thanks!

**Pull request merged**